tap_core/src/tap_manager/manager.rs

```diff
@@ -224,3 +245,21 @@ impl<
         ReceiptAggregateVoucher::aggregate_receipts(allocation_id, receipts, previous_rav)
     }
 }
+
+#[cfg(test)]
+pub mod manager_test_helpers {
+    use super::*;
+
+    /// Helper test function that gives access to the receipt storage
+    pub fn receipt_storage_adapter<'a, CA, RCA, RSA, RAVSA>(
+        manager: &'a mut Manager<CA, RCA, RSA, RAVSA>,
+    ) -> &'a mut RSA
+    where
+        CA: CollateralAdapter,
+        RCA: ReceiptChecksAdapter,
+        RSA: ReceiptStorageAdapter,
+        RAVSA: RAVStorageAdapter,
+    {
+        &mut manager.receipt_storage_adapter
+    }
+}
```

Comment:
If we are going to be accessing private members in tests there is an approach that allows for test to be separate without needing to add access helpers:

https://users.rust-lang.org/t/should-unit-tests-really-be-put-in-the-same-file-as-the-source/62153/2

Reply:
Yeah, I was actually wondering why those tests weren't there in the first place. If I want to avoid code duplication, I have to move pretty much all the tests, I think.

Reply:
I personally don't like the rust approach of putting unit tests in the source file. It is fine for small amount of unit test, but when it is 500+ lines of tests it feels like it clutters things a bit. You don't have to move the tests, you just add the test mod into `manager.rs`, then move the test file in the same directory as manager_test:

#[cfg(test)]
#[path = "manager_test.rs"]
mod manager_test;

I'll add a PR doing this and you can see if it's agreeable.

Reply:
@aasseman #135 This is what I was talking about. alternatively we could move all the unit tests into the same file. The current approach is just due to my personal preference, but either way would work.

Reply:
I like your solution!